data-prepper-plugins/csv-processor/src/main/java/com/amazon/dataprepper/plugins/processor/csv/CSVProcessor.java

```diff
@@ -12,22 +12,88 @@
 import com.amazon.dataprepper.model.processor.AbstractProcessor;
 import com.amazon.dataprepper.model.processor.Processor;
 import com.amazon.dataprepper.model.record.Record;
+import com.fasterxml.jackson.databind.MappingIterator;
+import com.fasterxml.jackson.dataformat.csv.CsvMapper;
+import com.fasterxml.jackson.dataformat.csv.CsvParser;
+import com.fasterxml.jackson.dataformat.csv.CsvSchema;
+
+import java.io.IOException;
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.List;
 
 /**
  * Processor to parse CSV data in Events.
  *
  */
 @DataPrepperPlugin(name="csv", pluginType = Processor.class, pluginConfigurationType = CSVProcessorConfig.class)
 public class CSVProcessor extends AbstractProcessor<Record<Event>, Record<Event>> {
+    private final List<String> header;
+    private final String quote_char;
+    private final String delimiter;
+    private final String source_key;
+    private final String headerSourceKey;
+    private final boolean hasUserSpecifiedHeader;
+    private final boolean hasHeaderSource;
+    private final Boolean isDeleteHeaderAfterProcessing;
+
     @DataPrepperPluginConstructor
     public CSVProcessor(final PluginMetrics pluginMetrics, final CSVProcessorConfig config) {
         super(pluginMetrics);
-        throw new UnsupportedOperationException();
+        this.source_key = config.getSource();
+        this.header = config.getColumnNames();
+        this.quote_char = config.getQuoteCharacter();
+        this.delimiter = config.getDelimiter();
+        this.headerSourceKey = config.getColumnNamesSourceKey();
+        this.isDeleteHeaderAfterProcessing = config.isDeleteHeader();
+
+        this.hasUserSpecifiedHeader = (this.header != null);
+        this.hasHeaderSource = (this.headerSourceKey != null);
     }
 
     @Override
     public Collection<Record<Event>> doExecute(final Collection<Record<Event>> records) {
+        CsvMapper mapper = createCsvMapper();
+        CsvSchema schema = createCsvSchema();
+
+        for (Record<Event> record : records) {
+
+            Event event = record.getData();
+
+            String message = event.get(this.source_key, String.class);
+            boolean this_event_has_header_source = this.hasHeaderSource && event.containsKey(this.headerSourceKey);
+
+            try {
+                MappingIterator<List<String>> message_it = mapper.readerFor(List.class).with(schema).readValues(message);
```

Comment:
`messageIterator` would be a better name here since the abbreviation "it" could lead to some confusion

Reply:
Agreed. I'll change `header_it` to `headerIterator` as well.